### PR TITLE
Add release notes for Java SDK 3.46.0

### PR DESCRIPTION
**Pull request by Emdee89**
## What Has Changed?

Explain what you have changed.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

**Review by newtork — APPROVED** [review]

File: docs/java/release-notes.mdx [hunk 1]
```diff
@@ -54,6 +54,62 @@ To continue using the latest features outlined in the release notes for version
...
 
+## 3.46.0 - June 3, 2021
+
+[Maven Central](https://search.maven.org/search?q=g:com.sap.cloud.sdk*%20AND%20v:3.46.0) | [Javadoc](https://help.sap.com/doc/55bb1642ea614bf582d514ed34401136/1.0/en-US/index.html)
+
+### Compatibility Notes
+
+- We updated the (OData V2) Trial Balance - Read (`C_TRIALBALANCE_CDS`) VDM to the [newest release 2105 of SAP S/4HANA Cloud](https://news.sap.com/2021/05/sap-s4hana-cloud-release-2105/)
+  
+  **Caution:** The service introduced some intended **breaking changes** with the latest update which are now, in consequence, also present in the VDM.
```
Comment (artemkovalyov) [inline docs/java/release-notes.mdx:65]:
```suggestion
- We updated the (OData V2) Trial Balance - Read (`C_TRIALBALANCE_CDS`) VDM to the [newest release 2105 of SAP S/4HANA Cloud](https://news.sap.com/2021/05/sap-s4hana-cloud-release-2105/)
  
  **Caution:** The service introduced some intended **breaking changes** with the latest update which are now, in consequence, also present in the type-safe client (VDM) by the SDK.
```

File: docs/java/release-notes.mdx [hunk 2]
```diff
@@ -54,6 +54,62 @@ To continue using the latest features outlined in the release notes for version
...
+  Furthermore, if a function is marked by the service as `composable` you can also apply an operation to the result of a function directly:
+  ```java
+  service.forEntity(person)
+        .withFunction(Person.getBestFriend())
+        .navigateTo(Person.FRIENDS)
+        .getAll();
+  ```
+  
+  The creation of the new operation types is enabled in the OData V4 generator as well.
+  Check out the full documentation of the feature for more information.
```
Comment (artemkovalyov) [inline docs/java/release-notes.mdx:90]: I couldn't find the related docs. Can we provide a link here?

**Review by artemkovalyov — CHANGES_REQUESTED** [review]
Mostly LGTM, but some minors need to be tackled.

File: docs/java/release-notes.mdx (hunk 2 above)
Comment (Emdee89) [reply docs/java/release-notes.mdx:90]: @artemkovalyov I am afraid this docu is still in progress, see https://github.com/SAP/cloud-sdk/pull/284

Comment (artemkovalyov) [reply docs/java/release-notes.mdx:90]: Yep, I found the PR. We can see if we can merge it by Thu.

**Comment by MatKuhr** [comment]
Could you please also merge #284 when merging this?

File: docs/java/release-notes.mdx (hunk 2 above)
Comment (MatKuhr) [reply docs/java/release-notes.mdx:91]: yeah if we merge the PR first we can link to the docs here.

**Review by artemkovalyov — APPROVED** [review]
LGTM

**Pull request merged**